replayserver/server/connection.py

```diff
@@ -49,12 +51,28 @@ async def write(self, data):
             raise MalformedDataError(f"Connection error: {short_exc(e)}")
         return True
 
-    def close(self):
+    def close(self, immediate=False):
+        if immediate:
+            self._do_close()
+        elif self._closing_coro is None:
+            self._closing_coro = asyncio.ensure_future(self._delayed_close())
```

Comment:
Technically its a `Task` not a coroutine object so `_closing_task` might be a better name.